### DISCUSSION TITLE
More options for transitioning between .mix files

### DISCUSSION THREAD

**Discussion by gmschroeder** [brunoherbelin/vimix#129] [Ideas]
Thank you for your receptivity and kindness! This is a great piece of software and a really cool person putting it together. I've found myself missing one thing now that I have a sense of how Vimix works.

Transitions! The cross-fades are great, and serve me for most circumstances. Expanding the options for changing between sessions would give me a lot more creative control.
Like the filters for sources, a handful of common choices with a "custom shader" option would meet my needs and then some.

In the transition view the options for timing and the like are only available in a sidebar menu (for me) now. That looks different from the video example you shared in the OSC bug thread. Are the transition options ultimately going to the view or the menu? This is what I'm seeing rn. To change transition settings I need to open that menu.

<img width="1023" alt="Screenshot 2024-03-10 at 9 28 03 AM" src="https://github.com/brunoherbelin/vimix/assets/2408592/5429a839-b2f8-4d67-b4f5-fd77af8422b8">

**Comment by brunoherbelin** [comment]
> Transitions! The cross-fades are great, and serve me for most circumstances. Expanding the options for changing between sessions would give me a lot more creative control. Like the filters for sources, a handful of common choices with a "custom shader" option would meet my needs and then some.

Okay, that could indeed be done
And thanks for the link to the [ShaderToy code](
https://www.shadertoy.com/view/NdGfzG) : this would definitely be a great source of inspiration.


> In the transition view the options for timing and the like are only available in a sidebar menu (for me) now. That looks different from the video example you shared in the OSC bug thread. Are the transition options ultimately going to the view or the menu? This is what I'm seeing rn. To change transition settings I need to open that menu.

This is definitely a bug :unamused: 

From screenshot, it looks like you use the OSX binary. Which version do you use?

**Comment by gmschroeder** [comment]
I'm running 0.82b.
The text is back in the transition view now? I figured I flipped a preference switch somewhere. Guess not. :p

I really like the Play&Open button in the sidebar. Is there anything keeping it out of the view?

**Comment by brunoherbelin** [comment]
> I really like the Play&Open button in the sidebar. Is there anything keeping it out of the view?

Doesn't it work by clicking on the target (right) source ?

**Reply by gmschroeder** [reply]
Clicking on 'open' or the target source in my install crossfades the .mix files and opens the mixing view. But crossfades really really quickly.
The "Play & Open" button in the transition sidebar works as expected.

**Reply by brunoherbelin** [reply]
and clic on the target itself? (magenta border)

This shows that the GUI is not clear... I will change that...

**Reply by gmschroeder** [reply]
Yep, that works. Nope, that isn't clear.

**Reply by brunoherbelin** [reply]
in the original version, there was no button, just the right target : so spontaneously it was the only thing to clic :) 
Also, did you try to grab and move the new source from left to right? (i.e. manual control of transition) ? 
What messed up is that I added the `Cancel` and `Open` buttons : so the GUI is now mixed and confusing... 

The goal being to offer direct 'hands-on' manipulation, how would you improve this transition view ?

**Reply by gmschroeder** [reply]
hmmm... 
I find the entire transition view opaque. I was relieved when I found the transition menu. I could 'play and open' the new .mix file. Which is all I wanted in the first place! I was annoyed that I needed another click, no matter what I wanted my transition to look like.
I didn't mess with the playlists (sessions?) in general for a long time. I didn't move the .mix file from left to right manually for a very, very long time. Maybe until I saw it in a picture? Even then, it seemed kind of silly.
I did not figure out clicking the thumbnail video (?) from the new .mix on my own. I was going to ask for the transition menu to be open when you click on a new .mix. The timeline felt stapled on and in the way. I planned to ignore the timeline if I could transition from the menu.
Why not put 'play&open' where play is if you really need the timeline? You'd still have the click and drag, but also one-click play&open and 'quick open.'
When would someone need 'play' but not 'open'?

**Comment by brunoherbelin** [comment]
Okay, Beta 7caded7c95ccf0ac6ccfe59d6ba066cbe795eeae has a work in progress improvement of Transition view.

![image](https://github.com/brunoherbelin/vimix/assets/60393189/ac39156d-7085-4749-9c49-07b25f7c911c)

Agreed that play animation is not useful without opening, and the 'open' button was not clear so now called 'fast open' with fast-forward icon to show the difference with play icon. Added play icon on the target to indicate to clic. Is it more clear now?

**Reply by gmschroeder** [reply]
That's a lot clearer. Is there something really important about clicking on the frame of the target video? Just wondering.

**Reply by brunoherbelin** [reply]
well, the magenta frame represents the output of vimix, so the metaphor is to bring the new session into the output. The click on the output is for simplicity and to be a big button for the most obvious thing do to.

But as you are suggesting more types of transitions, I am imagining that the layout could be different, and for example with multiple paths (eg. more than one line) that user can choose to go from session to output...

**Reply by gmschroeder** [reply]
I like the single line layout. But more wipes/fades/whatever in the drop-down and changing icons under the timeline. How do you see the multiple timelines thing working?

<img width="435" alt="Screenshot 2024-03-14 at 7 53 33 PM" src="https://github.com/brunoherbelin/vimix/assets/2408592/7dfc9798-0fa4-4e28-b1fb-1ef78939c417">

And be opinionated with the choices of transitions, pretty please. I know you dislike writing them, so frustrate someone with your choices. :-)

**Reply by zcbernard** [reply]
Hello, I like this new presentation of the sequence of sessions. A marginal question: why not put the "play" icon on the session to be launched rather than on the one already open, the same for the immediate opening icon under the session to be launched?
Happy end of weekend
Bernard

**Reply by brunoherbelin** [reply]
Love it! Why didn't I think of this earlier ? :wink: 

Please check if Beta b6213e1ed8d3fb81e838a40ed3a865227c19d0e6 is behaving as expected, but it should be as you suggest: an icon to play the animation on the candidate session, and an icon to fast forward on the target.

**Comment by zcbernard** [comment]
👍👍😀

**Comment by zcbernard** [comment]
Hello, in my opinion the two actions Play and Fast forward concern the
session to be launched. I will propose both under the session to be
launched. This also allows you to better visualize what is happening for
each session and to better choose the launch time. But these are really
just marginal remarks. Otherwise, I am finishing a video which summarizes
image training, as well as a proposed approach to handling Vimix. I will
submit this within the week.
I really like this software which, "less is more", does a lot of things,
doesn't offer anything superfluous, and requires us to rack our brains
sometimes to get around the problems posed by our desires.
Good day
Bernard
